core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -1132,18 +1167,19 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
 
             case async: Async[Any, Any, Any] =>
               updateLastTrace(async.trace)
-              self._blockingOn = async.blockingOn
-
               cur = initiateAsync(async.registerCallback)
 
               if (cur eq null) {
                 cur = drainQueueAfterAsync()
               }
 
               if (cur eq null) {
+                self._blockingOn = async.blockingOn
                 return null
               }
 
+              self._asyncContWith = AsyncContWith.`null`
```

Comment:
This was being reset in `initiateAsync` but not in `drainQueueAfterAsync`. I think this might explain hanging of TestClock reported by users as we were using `_asyncContWith eq null` to determine whether the fiber was suspended